internal/worker/modelworkermanager/modelworkermanager.go

```diff
@@ -203,7 +197,22 @@ func (m *modelWorkerManager) Wait() error {
 }
 
 func (m *modelWorkerManager) loop() error {
-	watcher := m.config.ModelWatcher.WatchModels()
+	controllerDomainServices := m.config.ControllerDomainServices
+	controllerModelUUID, err := controllerDomainServices.Controller().ControllerModelUUID(context.Background())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	domainServicesGetter := m.config.DomainServicesGetter
+
+	domainServices, err := domainServicesGetter.ServicesForModel(context.Background(), controllerModelUUID)
```

Comment:
This isn't quite right. It should be:
`watcher, err := m.config.ControllerDomainServices.Model().WatchActivatedModels(context.Background())`

Also delete the controller UUID bit.

Reply:
Thanks for pointing this out ian!